Recurring event module

AlehBarushka opened the pull request:
With the ability to set a recurring event (every day, on selected days of the week).

AlehBarushka committed:
added date utils

- "timeValidator" for checking the specified time;
- "setTime" for setting time and parses the reminder date.

AlehBarushka committed:
added reminder utils

- "reminderValidator" for validating reminder object;
- "dailyLoop" for repeating the reminder every day.

AlehBarushka committed:
added export

AlehBarushka committed:
changed script import

AlehBarushka committed:
added a reminder module

with the function of adding a reminder with a repeat for each day

AlehBarushka committed:
updated dateUtils

- updated 'setTime';
- added 'getCurrentWeekDay'.

AlehBarushka committed:
renamed file

AlehBarushka committed:
Create daysOfWeek-utils.js

added 'daysOfWeekСonverter' util

AlehBarushka committed:
Update Reminder.js

some renaming

AlehBarushka committed:
renamed and some refactoring

AlehBarushka committed:
updated interval

AlehBarushka committed:
added the ability to repeat reminders by day of the week

AlehBarushka committed:
Update reminder-utils.js

added jsDoc description "loop"

AlehBarushka committed:
added some array checking with days of the week

AlehBarushka committed:
deleted export instance

AlehBarushka committed:
added callback arg validation

AlehBarushka committed:
module renaming

AlehBarushka committed:
Update daysOfWeek-utils.js

vitali-working commented on module/RecurringEvent.js:
```diff
@@ -0,0 +1,86 @@
+import {
+  daysOfWeekValidator,
+  daysOfWeekСonverter,
+} from '../utils/daysOfWeek-utils.js';
```
I think  postfix `-utils` is redundant

vitali-working commented on utils/recurringEvent-utils.js:
```diff
@@ -0,0 +1,130 @@
+ * recurringEventValidator({title: 'meeting'})
+ **/
+export const recurringEventValidator = (event) => {
+  if (event?.title && event.time) {
```
Do you really need check `time` here?

vitali-working commented on utils/daysOfWeek-utils.js:
```diff
@@ -0,0 +1,50 @@
+ * // returns Error
+ * daysOfWeekValidator('Monday')
+ **/
+export const daysOfWeekValidator = (daysOfWeek) => {
```
A would say it's a bit strange validation. I think you can extend validation and add max length for array(array.length <= 7).

newqwes replied:
The name does not match, this is rather a check for an array and not a validator for `daysOfWeekValidator`

vitali-working commented on utils/date-utils.js:
```diff
@@ -0,0 +1,52 @@
+export const setTime = (time) => {
+  const currentDate = new Date();
+
+  const arrOfSettedTime = time.split(':');
```
I guess is more readable
`const [hours, minutes, seconds] = time.split(':');`

newqwes commented on utils/daysOfWeek-utils.js:
```diff
@@ -0,0 +1,50 @@
+ * daysOfWeekСonverter(['Monday', 'Tuesday', 'Sunday'])
+ **/
+export const daysOfWeekСonverter = (daysOfWeek) => {
+  const days = [
```
Move it separately to the constants file

newqwes commented on utils/daysOfWeek-utils.js:
```diff
@@ -0,0 +1,50 @@
+    'Saturday',
+  ];
+
+  let arrOfDayNumbers = [];
```
Here it is better to using reduce method

newqwes commented on utils/recurringEvent-utils.js:
```diff
@@ -0,0 +1,130 @@
+ * // returns 12
+ * loop(() => console.log('Hello'), 10000, 86400000, 'f9ca-1baa-c970-35b4')
+ **/
+const loop = (callback, delay, interval, eventId) => {
```
4 function parameters is a lot, it is better to pass no more than two, the rest if more then an object

newqwes commented on utils/recurringEvent-utils.js:
```diff
@@ -0,0 +1,130 @@
+ * dailyLoop(() => console.log('Hello'), '12:00:00')
+ **/
+export const dailyLoop = (callback, time) => {
+  const DAY_IN_MILLISECONDS = 86400000;
```
Move to variable file

newqwes commented on utils/recurringEvent-utils.js:
```diff
@@ -0,0 +1,130 @@
+ * dailyLoop(() => console.log('Hello'), '12:00:00', ['Monday', 'Tuesday'])
+ **/
+export const daysOfWeekLoop = (callback, time, daysOfweek) => {
+  const WEEK_IN_MILLISECONDS = 604800000;
```
Move to variable file

newqwes commented on utils/date-utils.js:
```diff
@@ -0,0 +1,52 @@
+ * @description The function returns the current day of the week.
+ * @returns {Number} returns the current day of the week according to local time, where 0 represents Sunday.
+ **/
+export const getCurrentWeekDay = () => {
```
redundant method

newqwes commented on utils/date-utils.js:
```diff
@@ -0,0 +1,52 @@
+ * @param {string} time - time of recurring event in format '15:00:00'.
+ * @returns {Number} returns the set date in milliseconds.
+ **/
+export const setTime = (time) => {
```
avoid code duplication. More or less like this:
```
export const getTimeInMillisecond = (time) => {
  const splittedTime = time.split(':')

  const timeInMillisecond = new Date().setHours(...splittedTime)

  return timeInMillisecond
}
```

AlehBarushka replied:
can I pass parameters as strings to the setHours method?
I checked, it works) 
but still, is it right?

AlehBarushka committed:
Updated checking of the array of days of the week

Moved 'days' separately to the constants file, changed checking the array of days of the week

AlehBarushka committed:
deleted 'getCurrentWeekDay'

AlehBarushka committed:
Moved constants to a variable file

AlehBarushka committed:
Update daysOfWeek.js

AlehBarushka committed:
Update daysOfWeek-utils.js

AlehBarushka committed:
updated the method of converting time to milliseconds

AlehBarushka committed:
deleted postfix -utils

AlehBarushka committed:
updated the function parameters

AlehBarushka committed:
Update daysOfWeek.js

AlehBarushka committed:
Update recurringEvent.js

AlehBarushka committed:
moved the time validation to the event object validation function

AlehBarushka committed:
renamed the file with constants

AlehBarushka committed:
changed the validator of the days of the week

AlehBarushka committed:
some fixes

AlehBarushka committed:
updated 'daysOfWeekConverter'

AlehBarushka committed:
some fixes

AlehBarushka committed:
Update daysOfWeek.js

AlehBarushka committed:
installed ESLint and some fixes
